### PR TITLE
rule for checking naming style

### LINKED ISSUE #328: Add prexif Check for inout ports io_

### PR DESCRIPTION
**Pull request by Nik-Sch**
closes #328 and #307

### CODE REVIEW

File: lib/rules/rulesBase.ts [hunk 1]
```diff
@@ -15,4 +17,34 @@ export class RuleBase {
...
+  if (token.type === TokenType.implicit) {
+    return;
+  }
+  let newName = token.text;
+  if (prefix.trim().length !== 0 && newName.startsWith(prefix) === false) {
+    newName = `${prefix}${newName}`;
+  }
+  if (suffix.trim().length !== 0 && newName.endsWith(suffix) === false) {
+    newName = `${newName}${suffix}`;
+  }
```
Comment (g0t00) [inline lib/rules/rulesBase.ts:33]: I like trimming the pre/suffix but should be done always? So `${prefix.trim()}${newName}`.
Also I am wondering should we make a settings normalization funktion or sth that does this automatically?
Doing this manually everywhere seems annoying?

File: package.json [hunk 2]
```diff
@@ -255,6 +249,18 @@
...
+					"type": "string",
+					"default": "",
+					"description": "Prefix for unused objects (no unused warning is shown for these objects)"
+				},
+				"VhdlLinter.style.unusedSuffix": {
+					"scope": "window",
+					"type": "string",
+					"default": "",
+					"description": "Prefix for unused objects (no unused warning is shown for these objects)"
+				},
```
Comment (g0t00) [inline package.json:263]: We had unused suffix before, can we keep this as a default?

File: package.json [hunk 3]
```diff
@@ -279,29 +309,53 @@
...
+					"description": "A Prefix for inouts (can be left empty)."
 				},
-				"VhdlLinter.ports.outRegex": {
+				"VhdlLinter.style.inoutSuffix": {
 					"scope": "window",
 					"type": "string",
-					"default": "^o_",
-					"description": "A regular expression to match output ports."
+					"default": "",
+					"description": "Suffix for inouts (can be left empty)."
```
Comment (g0t00) [inline package.json:346]: maybe also keep the port defaults?

File: test/test.ts [hunk 4]
```diff
@@ -61,11 +61,21 @@ async function run_test(path: URL, error_expected: boolean, projectParser?: Proj
...
+        },
+        style: {
+          unusedSuffix: '_unused'
+        },
+      })
+      : defaultSettingsWithOverwrite({
+        style: {
+          unusedSuffix: '_unused'
+        },
+      });
```
Comment (g0t00) [inline test/test.ts:78]: I think we test should be run at least with default settings.

File: package.json (hunk 2 above)
Comment (Nik-Sch) [reply package.json:263]: I am not sure about default values in general. I would prefer the default settings to be consistent and, therefore, I removed the unused suffix. But maybe we want to default to a suffix for all styles and then maybe add a hint towards the settings in the namingStyle rule?

Comment (g0t00) [reply package.json:263]: In general I think the default should interfere with existing code bases as little as possible while providing as much functionality as possible.
So having a default unused suffix/prefix is good I think (somebody randomly using _unused is unlikely).
But probably we should use empty defaults for the remaining styles. Or is there some existing style recommendation similar to pep8 or sth?

File: package.json (hunk 3 above)
Comment (g0t00) [reply package.json:346]: thinking about it no default is probably good. Even though I dislike suddenly dropping default functionality.

**Review by g0t00 — APPROVED** [review]

**Pull request merged**